### PR TITLE
Add g:cmake_ctest_args

### PR DESCRIPTION
**Pull request by mark2185**
~Not complete, but~ good ~enough~ for review and discussion.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ The current version of the plugin supports next commands:
 
  - **`:CMake`** creates a build directory (if it is necessary) and generates cmake project.
  - **`:CMakeBuild`** builds current cmake project. The command allows to specify cmake target.
- - **`:CTest`** runs tests. The command allow to specify Ctest arguments
+ - **`:CTest`** runs tests. The command allows to specify CTest arguments. Running with a bang ignores g:cmake_ctest_args
```
Comment (ilyachur) [inline README.md:51]: I propose to have 2 items for `:CTest` and `:CTest!` as we already have for `:CMakeRun`

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ The current version of the plugin supports next commands:
 
  - **`:CMake`** creates a build directory (if it is necessary) and generates cmake project.
```
Comment (ilyachur) [inline README.md:49]: Could you update the document: https://github.com/ilyachur/cmake4vim/blob/master/doc/cmake4vim.txt

File: README.md [hunk 3]
```diff
@@ -82,6 +82,7 @@ Plugin supports special global variables which are allow to change behaviour of
  - **`g:cmake_build_executor`** allows to force set the build executor. Available values are 'job', 'dispatch', 'system' and ''. Default is empty.
  - **`g:cmake_vimspector_support`** enables generation and modification of [Vimspector](https://github.com/puremourning/vimspector) config file. Default is 0. **Attention! The support of Vimspector config is an experimental feature.**
  - **`g:cmake_variants`** enables predefined cmake build variants in the form of a dictionary, e.g. `{ 'Debug' : { 'cmake_build_type' : 'Debug', 'cmake_usr_args' : '-DCONAN_PATH=~/.conan' } }`
+ - **`g:cmake_ctest_args`** enables arguments for `ctest`, e.g. `[ '-j8', '--output-on-failure', '--verbose' ]`. Default is `[]`. If the user calls `:CTest -R "test-pattern"`, the `args` are inserted directly after `ctest`, before `-R` parameter.
```
Comment (ilyachur) [inline README.md:85]: You have added `g:cmake_ctest_args` as an array, but other options (`g:make_arguments`, `g:cmake_usr_args` and etc.) are strings, I propose to use the unified approach for all options.

File: README.md (hunk 2 above)
Comment (mark2185) [reply README.md:49]: Oh yeah, I wanted to ask you, is that document automatically generated or written by hand?

Comment (ilyachur) [reply README.md:49]: Yes, this document is written manually 😄 I don't know the way to do it automatically.

File: test/tests/build/ctest_run.vader [hunk 4]
```diff
@@ -166,3 +167,37 @@ Execute (CTest with custom settings):
...
+        silent CMake -DCUSTOM_OP=ON
+        Assert filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should be generated"
+        silent CMakeBuild
+        Assert filereadable("cmake-build-Release/app/test_app"), "app should be built"
+        Assert filereadable("cmake-build-Release/tests/unit_tests"), "unit_tests should be built"
+        Assert filereadable("cmake-build-Release/lib/libtest_lib.a"), "test_lib should be built"
+        let g:cmake_ctest_args = [ '-j2', '--output-on-failure', '--verbose' ]
+        silent CTest -R "non-existant_pattern"
+        silent copen
+        " TODO: how to check this? it works when build_executor is dispatch because it puts the command as the title
```
Comment (mark2185) [inline test/tests/build/ctest_run.vader:183]: @ilyachur any ideas for this?

**Review by ilyachur — APPROVED** [review]
LGTM

**Pull request merged**